exodus_lambda/functions/origin_request.py

```diff
@@ -100,11 +100,37 @@ def resolve_aliases(self, uri):
 
         return uri
 
+    def lookup_listing(self, uri):
+        self.logger.info("Handling listing request: %s", uri)
+
+        target = uri[: -len("/listing")]
+        listing_data = self.definitions.get("listing", {})
+
+        if not listing_data:
+            self.logger.info("No listing data defined")
+            return {"status": "404", "statusDescription": "Not Found"}
+
+        listing = listing_data.get(target)
+
+        if not listing:
+            self.logger.info("No listing found for '%s'", target)
+            return {"status": "404", "statusDescription": "Not Found"}
+
+        return {
+            "body": "\n".join(listing["values"]),
+            "status": "200",
+            "statusDescription": "OK",
+        }
+
     def handler(self, event, context):
         # pylint: disable=unused-argument
 
         request = event["Records"][0]["cf"]["request"]
         uri = self.resolve_aliases(request["uri"])
+
+        if uri.endswith("/listing"):
```

Comment:
I know we've had discussion about this before and I think we said that the lambda code should be the only way of handling a request for listing files.  However, now seeing it in the code it made me nervous as we are essentially reserving the filename "listing" here, to mean only this kind of file and making it impossible to use for anything else.

I wondered are we really sure that the 'listing' filename is unused for other purposes, even for "raw rsynced" directory trees like kickstarts, so I did some investigation.  Turns out the answer is: they are sometimes used for other things.

An example is that I found /content/dist/rhel/server/6/6.1/x86_64/kickstart/ResilientStorage/listing, which exists and has a list of RPMs rather than $releasever or $basearch values.

So, it looks like we can't unconditionally assume that all URLs ending with "/listing" need to be handled by this mechanism as we would then block the publish of files named "listing" for other purposes, including the existing content above.

Could we please change it so that, if lookup_listing didn't find an entry for that listing file, we proceed with the normal handling?